components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PatientFamilyMetadataProvider.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.studies.family.internal;
+
+import org.phenotips.data.Patient;
+import org.phenotips.entities.PrimaryEntity;
+import org.phenotips.entities.PrimaryEntityMetadataProvider;
+import org.phenotips.studies.family.Family;
+import org.phenotips.studies.family.FamilyRepository;
+
+import org.xwiki.component.annotation.Component;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+/**
+ * Metadata provider that reports the family that a patient belongs to.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Component
+@Named("patient/family")
+@Singleton
+public class PatientFamilyMetadataProvider implements PrimaryEntityMetadataProvider
+{
+    @Inject
+    private FamilyRepository familyRepository;
+
+    @Override
+    public Map<String, Object> provideMetadata(PrimaryEntity entity)
+    {
+        if (entity instanceof Patient) {
+            Family f = this.familyRepository.getFamilyForPatient((Patient) entity);
```

Comment:
Can you remind me what happens if the user can access the patient but not its family? Does the family repository return null here? Do we want to show this metadatum if the family is inaccessible or not? IIRC, in the UIwe say the patient has a pedigree but the user cannot see it.

Reply:
I think we have an assumption that if a user can at least view a patient the user should be able to view the family as well? This can be broken for technical reasons, but isnt this how things are supposes to work?

Reply:
@allasm https://phenotips.atlassian.net/browse/PT-3427 describes a scenario where one can access the patient but not the family. This is how things work now. Do you think the  behavior should be changed?

Reply:
With the current code the family ID is exposed, but no other information about the family. I think that's OK, isn't it? Without disclosing it, the following behavior will break:

The user edits another family and wants to link a patient to a node in this pedigree. When searching for that patient, the response also includes whether it is part of another family already. If it is, then it is not allowed to be added in the new family. So, if we don't disclose that it is part of a family already, we would allow moving the patient to a new family, which means removing it from the other family, which is something the current user should not be able to do.

Reply:
Okay.